### PR TITLE
fix(components): [el-cascader] search result switch not highlight

### LINKED ISSUE #4040: After searching for change in the [bug report] ([bug] cascade filterable) input box, open the selection panel again and highlight not updated
### Element Plus version
1.1.0-beta.24

### OS/Browsers version
chrome 95.0.4638.54

### Vue version
3.2.16

### Reproduction Link
https://codepen.io/pen/?&prefill_data_id=444e995d-1c6b-453a-8f87-93d5f76e3eb8

### Steps to reproduce
1. 选中某一项（文档示例选中 Guide / Disciplines / Consistency）
2. 搜索其他非选中的根级（搜索 Sketch）并选择
3. 再次打开下拉框 下拉框中的选中高亮不同步高亮依旧为（Guide / Disciplines  / ）

### What is Expected?
  英语不好 望见谅

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

fix #4040

### CODE REVIEW

File: packages/components/cascader-panel/src/index.vue [hunk 1]
```diff
@@ -87,8 +87,8 @@ export default defineComponent({
 
     const config = useCascaderConfig(props)
 
-    const store: Ref<Store> = ref(null)
-    const menuList = ref([])
+    const store: Ref<Nullable<Store>> = ref(null)
+    const menuList: Ref<any[]> = ref([])
```
Comment (JeremyWuuuuu) [inline packages/components/cascader-panel/src/index.vue:91]:
```suggestion
    const store = ref<Nullable<Store>>(null)
    const menuList = ref<any[]>([])
```

**Comment by JeremyWuuuuu** [comment]
@btea Your fix completely broke the key press for `enter`, when you navigate the list via keyboard and try to select an item via key `enter`, you should be able to do that with the enter key pressing, but your fix just broke the feature, I don't think the root cause it from here.

**Comment by btea** [comment]
@JeremyWuuuuu I merged the dev code, and now it's ok.

File: packages/components/cascader-panel/src/index.vue (hunk 1 above)
Comment (JeremyWuuuuu) [reply packages/components/cascader-panel/src/index.vue]: Next time remember resolve the comments that you fixed. ;)

**Review by jw-foss — APPROVED** [review]

**Pull request merged**